gui/wxpython/datacatalog/tree.py

```diff
@@ -298,90 +302,94 @@ def __init__(
+            errors = []
+            location_nodes = []
+            nlocations = len(locations)
+            grassdata_node = self._model.AppendNode(
+                parent=self._model.root,
+                label=grassdatabase,
+                data=dict(type='grassdb', name=grassdatabase))
+            for location in locations:
+                results[location] = dict()
+                varloc = self._model.AppendNode(
+                    parent=grassdata_node, label=location, data=dict(
+                        type='location', name=location))
+                location_nodes.append(varloc)
+                loc_count += 1
+
+                Debug.msg(
+                    3, "Scanning location <{0}> ({1}/{2})".format(location, loc_count, nlocations))
+
+                q = Queue()
+                p = Process(target=getLocationTree,
+                            args=(grassdatabase, location, q))
+                p.start()
+
+                queue_list.append(q)
+                proc_list.append(p)
+                loc_list.append(location)
+
+                proc_count += 1
+                # Wait for all running processes
+                if proc_count == nprocs or loc_count == nlocations:
+                    Debug.msg(4, "Process subresults")
+                    for i in range(len(loc_list)):
+                        maps, error = queue_list[i].get()
+                        proc_list[i].join()
+                        if error:
+                            errors.append(error)
+
+                        for key in sorted(maps.keys()):
+                            mapset_node = self._model.AppendNode(
+                                parent=location_nodes[i],
+                                label=key, data=dict(
+                                    type='mapset', name=key))
+                            self._populateMapsetItem(mapset_node, maps[key])
+
+                    proc_count = 0
+                    proc_list = []
+                    queue_list = []
+                    loc_list = []
+                    location_nodes = []
 
         if errors:
             wx.CallAfter(GWarning, '\n'.join(errors))
         Debug.msg(1, "Tree filled")
 
         self.RefreshItems()
 
-    def UpdateCurrentLocationMapsetNode(self):
-        self.current_location_node, self.current_mapset_node = \
-            self.GetCurrentLocationMapsetNode()
+    def UpdateCurrentDbLocationMapsetNode(self):
+        self.current_grassdb_node, self.current_location_node, self.current_mapset_node = \
+            self.GetCurrentDbLocationMapsetNode()
 
     def ReloadTreeItems(self):
-        """Reload locations, mapsets and layers in the tree."""
+        """Reload dbs, locations, mapsets and layers in the tree."""
         self._orig_model = self._model
         self._model.RemoveNode(self._model.root)
         self.InitTreeItems()
-        self.UpdateCurrentLocationMapsetNode()
+        self.UpdateCurrentDbLocationMapsetNode()
```

Comment:
Move this at the end of _initTreeItems just before RefreshItems so that is is properly refreshed.